pkg/genericreconciler/genericreconciler.go

```diff
@@ -104,6 +106,33 @@ func specDiffers(o1, o2 metav1.Object) bool {
 	return true
 }
 
+func isReferringSameObject(a, b metav1.OwnerReference) bool {
+	aGV, err := schema.ParseGroupVersion(a.APIVersion)
+	if err != nil {
+		return false
+	}
+	bGV, err := schema.ParseGroupVersion(b.APIVersion)
+	if err != nil {
+		return false
+	}
+	return aGV == bGV && a.Kind == b.Kind && a.Name == b.Name
```

Comment:
This check is only sufficient if the two objects is in the same namespace. I think checking the uid rather than the name would be better.

Reply:
I like the uid idea.
So does o1.uid == o2.uid => o1.name == o2.name && o1.kind == o2.kind && o1.namespace == o2.namespace ?

OwnerReferences are scoped to namespace and a namespace field doesn't exist in the OwnerReference.

Reply:
It's worth while to document here why we used group/version/kind/name instead of UID. It's hard to understand without the context in which the function is used, similar to Janet's comment below.

@janetkuo I'm aware of the same namespace restriction, but not the cluster scope. Is there documentation somewhere about cluster-scope? FWIW, [kubernetes api reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#ownerreference-v1-meta) only mentions namespace.

Reply:
@huyhg I cannot find any document for cluster-scoped owners. The closest I can find is this PR https://github.com/kubernetes/kubernetes/pull/57211 that fixes some bugs about this feature. @caesarxuchao @liggitt we need to document this behavior in https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/